src/service/okapi.service.ts

```diff
@@ -27,18 +27,19 @@ class OkapiService extends RestService {
           'Accept': 'application/json'
         }
       }, (error: any, response: any, body: any) => {
-        if (response && response.statusCode >= 200 && response.statusCode <= 299) {
+        if (!!response && response?.statusCode >= 200 && response?.statusCode <= 299) {
           let accessToken;
           let refreshToken;
+          const headers = !!response?.headers ? response?.headers : undefined;
 
-          if (!!response.headers['set-cookie']) {
+          if (!!headers['set-cookie']) {
             const matchAccess = /folioAccessToken=([^;\s]*)/gi;
             const matchRefresh = /folioRefreshToken=([^;\s]*)/gi;
-            const cookies = !!response.headers['set-cookie'] ? response.headers['set-cookie'] : undefined;
+            const cookies = !!headers['set-cookie'] ? headers['set-cookie'] : undefined;
```

Comment:
Again, I think we should consider whether it is a good practice to assign variables as undefined.